mediator/esmFldsExchange_nems_mod.F90

```diff
@@ -159,6 +165,16 @@ subroutine esmFldsExchange_nems(gcomp, phase, rc)
     call addmap(fldListFr(compocn)%flds, 'So_t', compatm, maptype, 'ofrac', 'unset')
     call addmrg(fldListTo(compatm)%flds, 'So_t', mrg_from=compocn, mrg_fld='So_t', mrg_type='copy')
 
+    ! temporary conditional to avoid conflicts of advertised fields
+    ! when waves are passing through connectors
+    if (is_local%wrap%comp_present(compwav)) then
```

Comment:
Actually, we were also using these type conditionals in the HAFS specific field exchange file. I think it would be nice to unify those files to single one in the future. I think the conditional could also include the name of the component.

Reply:
I think the use in hafs is not quite the same. For both hafs and cesm, there are two calls the fldsExchange, one for advertise and one for initialize. The conditionals I see in the hafs and cesm fldExchanges are related to that---in both cases at the initialize call you're doing the addmap,addmerge only if the field is in the expFB and also present in the impFB. I do think it would be nice to have a cleaner way of doing all that checking though. 

But for the nems modes, esmFieldsExchange_nems is only called once (for phase=advertise) and the addmap and addmerge are done then. There is no checking for whether the fields are present or not. I did consider this as an issue, and refactored the esmfFieldsExchange_nems to do both an advertise and an initialize phase like for hafs and cesm. But I still had the same issue---with waves going through the connectors, having the mediator also trying to advertise those same fields (even if they're not connected) broke reproducibility. So this conditional completely removes the advertisement of fields if the mediator is not in use for the waves.